python/pyxbos/pyxbos/drivers/hvac/hvac.py

```diff
@@ -0,0 +1,162 @@
+    def setup(self, cfg):
+        # A dictionary mapping point names to their classes
+        self.pymortar_client = pymortar.Client({
+            'username': cfg['mortar_api_username'],
+            'password': cfg['mortar_api_password']
+        })
+        self.bacnet = BAC0.connect()
+        self.point_class_map = get_point_class_mapping(self.pymortar_client, cfg['building'])
+        self.devices = [BAC0.device(dev[2], dev[3], self.bacnet) for dev in self.bacnet.devices]
+
+    def read(self, requestid=None):
+        for dev in self.devices:
+            # A dictionary mapping point names to their bacnet values
+            point_value_map = get_point_value_mapping(dev.points)
+
+            #AHU-1.DPR-O Damper point MISSING from Bacnet points
+
+            output = {}
+            for equipclass in self.point_class_map:
+                if equipclass not in ['Boiler', 'Chilled_Water_Pump', "Hot_Water_Pump"]:
+                    if equipclass not in output:
+                        output[equipclass] = {}
+
+                    for point in self.point_class_map[equipclass]:
+                        pieces = point.split(".")
+                        new_point = point
+                        if len(pieces) > 2:
+                            new_point = pieces[0] + "." + pieces[2]
+
+                        if "DPRPOS" in point:
+                            new_point = pieces[0] + "." + "DMPRPOS"
+
+                        if new_point in point_value_map:
+                            pt_class = self.point_class_map[equipclass][point]['point_class']
+                            if pieces[0] not in output[equipclass]:
+                                output[equipclass][pieces[0]] = {}
+                            output[equipclass][pieces[0]][pt_class.lower()] = point_value_map[new_point]
+
+            for equipclass in output:
+                data = {}
+                all_eq = output[equipclass]
+                for eq in all_eq:
+                    props = {}
+                    for prop in all_eq[eq]:
+                        if prop not in ['vvt', 'vav']:
+                            print("EQ", eq, prop)
+                            value = all_eq[eq].get(prop, None)
+                            if not isinstance(value, str):
+                                props[prop] = types.Double(value=value)
+                            else:
+                                if value == "Cooling":
+                                    props[prop] = 1
+                                else:
+                                    props[prop] = 0
+
+                    data[equipclass.lower()] = getattr(hvac_pb2, equipclass)(**props)
+                    data['time'] = int(time.time()*1e9)
+
+                    msg = xbos_pb2.XBOS(XBOSIoTDeviceState = iot_pb2.XBOSIoTDeviceState(**data))
+                    self.report(eq, msg) # Put name of equipment here as the first argument
+
+def get_point_value_mapping(device_points):
```

Comment:
Can you write some documentation explaining what this method is doing? It looks like it is just reading and parsing strings; is that how we get values out of BAC0?